include/revng/Pipeline/Rank.h

```diff
@@ -17,30 +18,96 @@ class Rank : public DynamicHierarchy<Rank> {
   Rank(llvm::StringRef Name, Rank &Parent) : DynamicHierarchy(Name, Parent) {}
 };
 
-  template<StringTemplateParameter Str, typename LocationType, typename ParentRnk>
-  class RankDefinition : public Rank {
+namespace detail {
+
+template<typename... TypesToAppend, typename... InitialTupleTypes>
+inline constexpr auto appendImpl(std::tuple<InitialTupleTypes...> &&Tuple) {
+  return std::tuple<InitialTupleTypes..., TypesToAppend...>();
+}
+
+template<typename Tuple, typename... TypesToAppend>
+struct AppendToTupleHelper {
+  using type = decltype(appendImpl<TypesToAppend...>(std::declval<Tuple>()));
+};
+
+template<typename First, typename Second>
+struct CH {
```

Comment:
Bad name

Reply:
80-character long lines are brutal 😅 

Will rename the struct and make an alias for it though. Sorry.

Reply:
Gone.